idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,361 @@
+                    b.solids[t].get_material_density_terms("Sol", j)
+                    for j in b.config.solid_property_package.component_list)
+
+        # Sum of mass fractions at all time equals 1
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Sum of mass fractions at all time")
+        def sum_component_constraint(b, t):
+            if t == b.flowsheet().config.time.first():
+                return Constraint.Skip
+            else:
+                return 1e2 == 1e2*sum(b.solids[t].mass_frac_comp[j]
+                                      for j in b.solids[t].
+                                      _params.component_list)
+
+        if self.config.energy_balance_type != EnergyBalanceType.none:
+            # Solid phase energy balance
+            # Accumulation equal to heat transfer
+            self.solids_energy_holdup = Var(
+                    self.flowsheet().time,
+                    initialize=1,
+                    doc="Solid phase energy holdup",
+                    units=pyunits.kJ)
+
+            @self.Constraint(self.flowsheet().config.time,
+                             doc="Solid phase energy holdup constraints")
+            def solids_energy_holdup_constraints(b, t):
+                return b.solids_energy_holdup[t] == (
+                      self.volume_solid[t] *
+                      b.solids[t].get_energy_density_terms("Sol"))
+
+            self.solids_energy_accumulation = DerivativeVar(
+                        self.solids_energy_holdup,
+                        wrt=self.flowsheet().config.time,
+                        doc="Solids energy accumulation",
+                        units=pyunits.kJ/pyunits.s)
+
+            @self.Constraint(self.flowsheet().config.time,
+                             doc="Solid phase energy accumulation constraints")
+            def solids_energy_accumulation_constraints(b, t):
+                return b.solids_energy_accumulation[t] == \
+                    - sum(b.reactions[t].reaction_rate[r] *
+                          b.volume_solid[t] *
+                          b.reactions[t].dh_rxn[r]
+                          for r in b.config.reaction_package.
+                          rate_reaction_idx)
+
+        if self.config.energy_balance_type == EnergyBalanceType.none:
+            # Fix solids temperature to initial value for isothermal conditions
+            @self.Constraint(
+                    self.flowsheet().config.time,
+                    doc="Isothermal solid phase constraint")
+            def isothermal_solid_phase(b, t):
+                if t == b.flowsheet().config.time.first():
+                    return Constraint.Skip
+                else:
+                    return (
+                            b.solids[t].temperature ==
+                            b.solids[0].temperature)
+
+    def initialize(blk, state_args=None, outlvl=idaeslog.NOTSET,
```

Comment:
Should there be separate state args for the gas and solids phases?

Reply:
@chineduobiora please comment on this, but I don't believe we need to keep `state_args` as an argument here. It doesn't appear to be used by the initialization routine, as the gas and solid phase flags are pre-defined earlier in the `build()` method using separate state args as inputs.

Reply:
I see that the other two CLC models pass `gas_phase_state_args` and `solid_phase_state_args` to the initialization routines for the gas and solid inlet blocks. I could add them to the flags statements as arguments, although the gas phase is only required for the reactions and has fixed properties.

Reply:
@bpaul4 yes there should be state args for the gas and solid phases just like for the other two gas solid contactor models. At the moment, the gas phase has fixed properties and is only required for the reactions, however the gas density/composition still need to be initialized from the fixed state properties.